corehq/ex-submodules/casexml/apps/case/cleanup.py

```diff
@@ -136,14 +137,20 @@ def claim_case(domain, restore_user, host_id, host_type=None, host_name=None, de
     return claim_id
 
 
-def get_first_claim(domain, user_id, case_id):
+def get_first_claims(domain, user_id, case_ids):
     """
-    Returns the first claim by user_id of case_id, or None
+    Returns the first claim by user_id of case_ids, or None
     """
-    case = CommCareCase.objects.get_case(case_id, domain)
-    identifier = DEFAULT_CASE_INDEX_IDENTIFIERS[CASE_INDEX_EXTENSION]
-    try:
-        return next((c for c in case.get_subcases(identifier)
-                     if c.type == CLAIM_CASE_TYPE and c.owner_id == user_id and c.closed is False))
-    except StopIteration:
-        return None
+    case_ids_found = CommCareCase.objects.get_case_ids_that_exist(domain, case_ids)
+    cases_not_found = [case for case in case_ids if case not in case_ids_found]
+
+    if len(cases_not_found) != 0:
+        raise CaseNotFound(", ".join(cases_not_found))
+
+    potential_cases = CommCareCase.objects.get_reverse_indexed_cases(
+        domain, case_ids_found, case_types=[CLAIM_CASE_TYPE], is_closed=False)
+    # creates set of claimed case_ids where owner_id = user_id
+    previously_claimed_ids = set(map(lambda case: case.get_index('host').referenced_id
+        if case.owner_id == user_id else None, potential_cases))
+
+    return previously_claimed_ids
```

Comment:
This will have `None` in the set, which isn't really a problem, but it's a bit weird.  In my opinion a set comprehension works a little smoother here

Reply:
okay went ahead and changed this to set comprehension

Reply:
thanks, looks good!